### DISCUSSION TITLE
A lightcull var.

### DISCUSSION THREAD

**Discussion by SniperGoth** [redeclipse/base#1126] [Ideas and Feature Requests]
So, there's this little var called /fogcullintensity in the game.
It basically culls everything out of the rendering view (models, lights etc), because things inside the fog don't really need to be rendered.
But since this specific var isn't fog-dependant, and can be used anywhere, it acts as a sort of an "performance lod" of sorts.
See examples below : 

/fogcullintensity 0.001 (default value)
![20180120233551](https://user-images.githubusercontent.com/24359960/35189862-ff303ff0-fe3a-11e7-9b65-0c92f574e7f8.png)

/fogcullintensity 0.6  (~10+fps gain)
![20180120233604](https://user-images.githubusercontent.com/24359960/35189863-ff55b708-fe3a-11e7-9c13-441cc363c9ab.png)

/fogcullintensity 0.001 
![20180120233634](https://user-images.githubusercontent.com/24359960/35189908-31d851b2-fe3c-11e7-91a4-e1b55acf31b6.png)

/fogcullintensity 0.6  (around 20+ fps gain on this specific place)
![20180120233628](https://user-images.githubusercontent.com/24359960/35189911-38f96ea4-fe3c-11e7-8566-96c76f916cf8.png)

But anyway, what i mean is, can we get a var that uses these properties, but culls out ONLY light entities?

People could even set these settings on the graphics menu, as it could be used as a massive performance increase property, depending on the map and it's cull intensity, on more lower-end systems.
The suggested settings could be : 

Light entity render distance:
Extreme = like it is right now, no culling.
High = around cullintensity 0.8
Medium = cullintensity 0.6 ~
Low = cullintensity 0.4 .

Is this an unreasonable request, or could it be implemented instead of we messing with random values ourselves with the /fogcullintensity var and risk getting invisible models and player characters in the process?

**Comment by qreeves** [comment]
I don't recommend culling anything unless it *absolutely* can't be seen. You'll get the same kind of problem; flickering lights, etc. You should always target performance to a base level and allow the user to change their *graphics settings* accordingly.

**Comment by SniperGoth** [comment]
Yeah, i know that we mappers should optimize our maps and such, but im asking this because of the massive performance boost it could give overall, as it's something to consider, since the suggestion is about letting the user use this if they wish, (More performance -/////- looking better) instead of making it a map-var.
We don't really want to see lots and lots of small lights being rendered at a very good distance if it's not necessary, so please, take this in consideration for implementation.

**Comment by qreeves** [comment]
I can't, even with Tesseract, the new engine isn't designed to "look better" than what we had before, it's more about using shaders and making small arena based maps that make good use of occlusion. The Cube Engine has never been about forms of LoD or anything like that, and you will never see it added.

To be honest, this isn't even a decision I can make, and you guys needs to start listening to the Lead Programmer more. It's not you specifically SniperGoth, but others in the community are failing to heed the advice he gives; something he's not obliged to do, and most likely regrets it when he does.

He has encouraged me the entire way for the past ten plus years, dedicated his free time to supporting this project, and *trying* to teach us all. If it weren't for him, there'd be no Cube Engine 2 as you know it today, no Tesseract, no Red Eclipse or Blood Frontier or even Sauermod. Don't alienate the person who's really your greatest ally in this programming world, provided you show him a bit of respect and listen to his reasoning.

I'm happy with the decision I made at the beginning to use the Cube Engine. Sure, I push it past it's limits sometimes, but I have learned more than I ever would have working with some other (and likely more complete) engine because there was always a challenge waiting to be met. We are unique.

I can see the frustration building on both sides, and I haven't been immune to this myself. I think this needs to be resolved before the tension breaks and chaos ensues. The only thing is to do is accept that we're different and use the engine the way it was intended, *please*. I can understand if people don't feel the same way we do, but there are other games on other engines if you're not happy with this one. I don't like having to say that, but it's the only real truth. Sorry.

**Comment by SniperGoth** [comment]
I know all the story leading up to the present moment, the struggles, the problems etc.
It's just that there are certain things that, if it could help performance for any system, even by a small bit, could be beneficial.
Sorry if i offended you and others, that wasn't my intention.

**Comment by qreeves** [comment]
Not offended, just taking the opportunity to be honest so everyone understands before it gets worse. ❤️
